homeassistant/components/meteo_france/__init__.py

```diff
@@ -1,121 +1,102 @@
-            cv.ensure_list,
-            [
-                vol.Schema(
-                    {
-                        vol.Required(CONF_CITY): cv.string,
-                        vol.Optional(CONF_MONITORED_CONDITIONS): vol.All(
-                            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-                        ),
-                    }
-                )
-            ],
-            has_all_unique_cities,
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [CITY_SCHEMA]))}, extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up the Meteo-France component."""
-    hass.data[DATA_METEO_FRANCE] = {}
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up Meteo-France from legacy config file."""
 
-    # Check if at least weather alert have to be monitored for one location.
-    need_weather_alert_watcher = False
-    for location in config[DOMAIN]:
-        if (
-            CONF_MONITORED_CONDITIONS in location
-            and "weather_alert" in location[CONF_MONITORED_CONDITIONS]
-        ):
-            need_weather_alert_watcher = True
+    conf = config.get(DOMAIN)
+    if conf is None:
+        return True
 
-    # If weather alert monitoring is expected initiate a client to be used by
-    # all weather_alert entities.
-    if need_weather_alert_watcher:
-        _LOGGER.debug("Weather Alert monitoring expected. Loading vigilancemeteo")
-
-        weather_alert_client = VigilanceMeteoFranceProxy()
-        try:
-            weather_alert_client.update_data()
-        except VigilanceMeteoError as exp:
-            _LOGGER.error(
-                "Unexpected error when creating the vigilance_meteoFrance proxy: %s ",
-                exp,
+    for city_conf in conf:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=city_conf.copy()
             )
-    else:
-        weather_alert_client = None
-    hass.data[DATA_METEO_FRANCE]["weather_alert_client"] = weather_alert_client
+        )
 
-    for location in config[DOMAIN]:
+    return True
 
-        city = location[CONF_CITY]
 
-        try:
-            client = meteofranceClient(city)
-        except meteofranceError as exp:
-            _LOGGER.error(
-                "Unexpected error when creating the meteofrance proxy: %s", exp
-            )
-            return
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up an Meteo-France account from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
 
-        client.need_rain_forecast = bool(
-            CONF_MONITORED_CONDITIONS in location
-            and "next_rain" in location[CONF_MONITORED_CONDITIONS]
+    # Weather alert
+    weather_alert_client = await hass.async_add_executor_job(VigilanceMeteoFranceProxy)
+    try:
+        weather_alert_client.update_data()
```

Comment:
I/O?

Reply:
Whoops wrong place : `VigilanceMeteoFranceProxy` is not doing I/O